server/datastore/mysql/migrations/tables/20240703193943_AddBundleIdentifierToSoftwareTitles.go

```diff
@@ -0,0 +1,92 @@
+package tables
+
+import (
+	"database/sql"
+	"fmt"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20240703193943, Down_20240703193943)
+}
+
+func Up_20240703193943(tx *sql.Tx) error {
+	if columnExists(tx, "software_titles", "bundle_identifier") {
+		return nil
+	}
+
+	_, err := tx.Exec(`
+		ALTER TABLE software_titles 
+		ADD COLUMN bundle_identifier varchar(255) COLLATE utf8mb4_unicode_ci DEFAULT NULL`)
+	if err != nil {
+		return fmt.Errorf("adding bundle_identifier to software_titles: %w", err)
+	}
+
+	_, err = tx.Exec(`
+	      UPDATE
+		software_titles st,
+		software s
+	      SET
+		st.bundle_identifier = NULLIF(s.bundle_identifier, '')
```

Comment:
Would it be simpler if the column was `NOT NULL DEFAULT ''`  or is the empty vs NULL distinction important? And use `COALESCE(s.bundle_identifier, '')` since software.bundle_identifier can be NULL?

Reply:
Ah I see you use `NULL` for the deletion below, feel free to ignore that comment.